build_web_compilers/lib/src/dev_compiler_builder.dart

```diff
@@ -88,42 +102,75 @@ Future createDevCompilerModule(Module module, BuildStep buildStep,
     request.arguments.add('--no-source-map');
   }
 
-  // Add the default analysis_options.
-  await scratchSpace.ensureAssets([defaultAnalysisOptionsId], buildStep);
-  request.arguments.add(defaultAnalysisOptionsArg(scratchSpace));
-
   // Add all the linked summaries as summary inputs.
   for (var id in transitiveSummaryDeps) {
-    request.arguments.addAll(['-s', scratchSpace.fileFor(id).path]);
+    if (useKernel) {
+      var path = p.url.relative(scratchSpace.fileFor(id).path,
```

Comment:
(optional as well): to make sharing of logic more clear, it might help to rewrite this as:
```dart
  var summaryPath = scratchSpace.fileFor(id).path;
  if (useKernel) {
    summaryPath = path.url.relative(summaryPath, from: ...);
  }
  request.arguments.addAll(['-s', summaryPath]);
```

Reply:
done, used ternary though